### PR TITLE
PR: Fix collision between `format_on_save` and other Editor formatting options

### LINKED ISSUE #17716: Files not autoformatted on save when other options that work on save are set
### What steps will reproduce the problem?

1. Turn on auto-format on save on, with black enabled. (haven't tested with autopep8)

2. create a new file, input poorly formatted code such as

    a = [3,
              4]

3. press save. 

4. inspect file in external editor to see it has not been formatted.

5. In spyder, press CTRL+ALT+I to format the file. You get the expected

    a = [3, 4]

formatting. This proves black is working and spyder knows how to invoke it correctly. 

Also, if you put several blank lines at the end of the file, when I press save the lines are correctly deleted, proving that format during save works in general (I have the check box for removing extra lines on save checked)

## Versions

* Spyder version: 5.3.0 
* Python version: 3.8.8
* Qt version: 5.12.10
* PyQt5 version: 5.12.3
* Operating System: Windows 7

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.31.1;<8.0.0       :  7.32.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.17.2 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  22.3.0 (OK)
nbconvert >=4.0               :  6.0.7 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
paramiko >=2.4.0              :  2.7.2 (OK)
parso >=0.7.0;<0.9.0          :  0.7.0 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.8.1 (OK)
pylint >=2.5.0                :  2.7.4 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.4.1;<1.5.0          :  1.4.1 (OK)
pylsp_black >=1.2.0           :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0     :  3.0.3 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=1.0.2             :  1.0.2 (OK)
qtconsole >=5.3.0;<5.4.0      :  5.3.0 (OK)
qtpy >=2.0.1                  :  2.0.1 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  52.0.0.post20210125 (OK)
sphinx >=0.6.6                :  4.0.1 (OK)
spyder_kernels >=2.3.0;<2.4.0 :  2.3.0 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  1.0.2 (OK)
zmq >=17                      :  22.3.0 (OK)

# Optional:
cython >=0.21                 :  0.29.23 (OK)
matplotlib >=3.0.0            :  3.3.4 (OK)
numpy >=1.7                   :  1.20.1 (OK)
pandas >=1.1.1                :  1.2.4 (OK)
scipy >=0.17.0                :  1.6.2 (OK)
sympy >=0.7.3                 :  1.8 (OK)
```

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

There was a collision between the autoformat on save option other Editor formatting options available. Due to that those options need to be disregard if autoformat on save is on.

### Issue(s) Resolved

Fixes #17716 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

File: spyder/plugins/editor/confpage.py [hunk 1]
```diff
@@ -148,6 +148,21 @@ def setup_page(self):
             'tab_stop_width_spaces',
             default=4, min_=1, max_=8, step=1)
 
+        format_on_save = CONF.get(
+            'completions',
+            ('provider_configuration', 'lsp', 'values', 'format_on_save'),
+            False
+        )
+        if format_on_save:
```
Comment (ccordoba12) [inline spyder/plugins/editor/confpage.py:156]: This is ok when preferences is opened and auto-format on save is already set, but not if you enable it and go to this tab in the editor. In that case, these options are still enabled (but they shouldn't).

To solve that, perhaps you could use `SpyderConfigurationObserver` here?

**Review by ccordoba12 — COMMENTED** [review]
Thanks @dalthviz for your work on this! I think it needs a bit more of work.

File: spyder/plugins/editor/confpage.py (hunk 1 above)
Comment (dalthviz) [reply spyder/plugins/editor/confpage.py:156]: I think that even with that the config page will not take into account tha change unless the user clicks at leats the Apply button, is that ok?

Comment (dalthviz) [reply spyder/plugins/editor/confpage.py:156]: Checking the usage of `SpyderConfigurationObserver` for some reason it gets stuck :/ Could we merge this as it is (to have the translation string to do the translation PR update) and check the missing UX behavior in a different following PR @ccordoba12 ?

**Review by ccordoba12 — APPROVED** [review]
I tested your latest changes locally and things are working as expected now. Thanks @dalthviz!

**Pull request merged**